src/modules/commerce_paymentwall/commerce_paymentwall.module

```diff
@@ -0,0 +1,282 @@
+    );
+
+    return $items;
+}
+
+function commerce_paymentwall_libraries_info()
+{
+    $libraries = array();
+    $libraries['paymentwall'] = array(
+        'name' => 'paymentwall',
+        'vendor' => 'http://www.paymentwall.com/',
+        'path' => 'paymentwall-php/lib',
+        'version' => '1.0.0',
+        'files' => array(
+            'php' => array(
+                'paymentwall.php',
+            ),
+        ),
+    );
+    return $libraries;
+}
+
+function commerce_paymentwall_init_config()
+{
+    $payment = commerce_payment_method_instance_load('commerce_paymentwall|commerce_payment_commerce_paymentwall');
+    $settings = $payment['settings'];
+    libraries_load('paymentwall');
+    Paymentwall_Config::getInstance()->set(array(
+        'api_type' => Paymentwall_Config::API_GOODS,
+        'public_key' => $settings['project_key'],
+        'private_key' => $settings['secret_key']
+    ));
+    return $settings;
+}
+
+function commerce_paymentwall_add_transaction($order,
+                                              $instance_id,
+                                              $amount,
+                                              $currency_code,
+                                              $status,
+                                              $message,
+                                              $remote_id)
+{
+
+    // Create a new transaction entity.
+    $transaction = commerce_payment_transaction_new('commerce_paymentwall', $order->order_id);
+    $transaction->instance_id = $instance_id;
+    $transaction->amount = $amount;
+    $transaction->currency_code = $currency_code;
+    $transaction->status = $status;
+    $transaction->message = $message;
+    $transaction->remote_id = $remote_id;
+
+    // Save the transaction.
+    commerce_payment_transaction_save($transaction);
+}
+
+function commerce_paymentwall_submit_form_validate($payment_method, $pane_form, $pane_values, $order, $form_parents = array())
+{
+    $widgetUrl = commerce_paymentwall_credit_card_validate($pane_values, $payment_method, $order, array('commerce_payment', 'payment_details'));
+    commerce_order_status_update($order, COMMERCE_PAYMENTWALL_ORDER_PROCESSING, false, true, 'Waiting payment via Paymentwall');
+    drupal_goto($widgetUrl);
+}
+
+function commerce_paymentwall_credit_card_validate($submitted_values, $payment_method_instance, $order = NULL, $form_parents = array())
+{
+    $balance = commerce_payment_order_balance($order);
+    $instance_id = $payment_method_instance['instance_id'];
+    $message = "Payment via Paymentwall";
+    $remote_id = $_SERVER['REMOTE_ADDR'];
+    $status = COMMERCE_PAYMENT_STATUS_SUCCESS;
+
+    commerce_paymentwall_add_transaction($order,
+        $instance_id,
+        $balance['amount'],
+        $balance['currency_code'],
+        $status,
+        $message,
+        $remote_id
+    );
+
+    return commerce_paymentwall_get_widget($order, $balance, $payment_method_instance['settings']);
```

Comment:
Please add more params to support User profile API, and `integration_module=drupal-commerce`